site/static/api/device-registry-revised.yaml

```diff
@@ -0,0 +1,713 @@
+openapi: 3.0.1
+info:
+   title: Eclipse Hono™ Device Registry API
+   description: |
+      This API allows one to interact with the Eclipse Hono Device Registry
+      API. It acts as a common basis which all Hono device registries should
+      implement.
+
+      All operations, except the `tenants` resource are required. The tenant
+      management might be outside of the scope of the device registry and
+      managed by a higher level system. In this case all calls should simply
+      return `404`. However, if the `tenants` resource is implemented, then all
+      operations of it must be implemented.
+   contact:
+      email: "hono-dev@eclipse.org"
+   license:
+      name: EPL-2.0
+      url: https://www.eclipse.org/legal/epl-2.0/
+   version: 1.0.0
+
+externalDocs:
+   description: Eclipse Hono™ homepage
+   url: https://eclipse.org/hono
+
+tags:
+   - name: tenants
+     description: Tenant Management (optional)
+     externalDocs:
+        description: Hono Multi-Tenancy
+        url: https://www.eclipse.org/hono/concepts/tenancy/
+   - name: devices
+     description: Device registration
+     externalDocs:
+        description: Hono device identity
+        url: https://www.eclipse.org/hono/concepts/device-identity/
+   - name: credentials
+     description: Device credentials
+     externalDocs:
+        description: Hono device identity
+        url: https://www.eclipse.org/hono/concepts/device-identity/
+
+servers:
+   - url: '{corsProxy}{server}/v1'
+     variables:
+        server:
+           default: http://hono.eclipse.org:28080
+        corsProxy:
+           default: https://cors-anywhere.herokuapp.com/
+
+security:
+   - BearerAuth: []
+
+paths:
+
+# Tenant API
+
+   /tenants:
+
+      post:
+         tags:
+            - tenants
+         summary: Create new tenant with auto-generated ID
+         operationId: createTenant
+         requestBody:
+            description: New tenant information
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Tenant'
+            required: true
+         responses:
+            201:
+               $ref: '#/components/responses/Created'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            409:
+               $ref: '#/components/responses/AlreadyExists'
+
+   /tenants/{tenant}:
+
+      post:
+         tags:
+            - tenants
+         summary: Create new tenant
+         operationId: createTenantWithId
+         parameters:
```

Comment:
Even thought this naming is valid in terms of rest and openapi in general, its problematic for most of the open-api code generators(at least the java once). Since both, the path-parameter and the body are called "tenant", the generated interfaces would have duplicate variable names in the updateTenant and createTenantWithId methods. 
I would suggest to call the pathParameter something like tenantName or tenantId (similar to deviceId in the DevicesApi) to avoid such problems.

Reply:
Interesting point, I just tried that:

~~~
$ npx @openapitools/openapi-generator-cli generate -i device-registry-revised.yaml -g java -o build/
npx: installed 1 in 1.252s
Exception in thread "main" org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 9, Warning count: 0
Errors: 
	-attribute paths.'/devices/{tenant}/{deviceId}'(put).parameters.There are duplicate parameter values
	-attribute paths.'/credentials/{tenant}/{deviceId}'(get).parameters.There are duplicate parameter values
	-attribute paths.'/devices/{tenant}/{deviceId}'(delete).parameters.There are duplicate parameter values
	-attribute paths.'/devices/{tenant}/{deviceId}'(get).parameters.There are duplicate parameter values
	-attribute paths.'/credentials/{tenant}/{deviceId}'(put).parameters.There are duplicate parameter values
	-attribute paths.'/credentials/{tenant}/{deviceId}'(delete).parameters.There are duplicate parameter values
	-attribute paths.'/tenants/{tenant}'(put).parameters.There are duplicate parameter values
	-attribute paths.'/tenants/{tenant}'(delete).parameters.There are duplicate parameter values
	-attribute paths.'/devices/{tenant}/{deviceId}'(post).parameters.There are duplicate parameter values

	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:626)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:367)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:60)
~~~

So that is indeed something that needs fixing. Thanks!

Reply:
Ok, I did clean up the parameter definitions. However I think the remaining validation errors are due to:

* https://github.com/OpenAPITools/openapi-generator/issues/2631
* https://github.com/OpenAPITools/openapi-generator/pull/2775

However that shouldn't be a real problem. I will check the body issue now.

Reply:
Ok, I see your point now. Code generation is always fun :roll_eyes:

Reply:
Should be fixed now.

Reply:
Yes, looks good now.:)